datasetinsights/io/downloader/gcs_downloader.py

```diff
@@ -0,0 +1,30 @@
+from datasetinsights.io.downloader.base import DatasetDownloader
+from datasetinsights.io.gcs import download_folder_from_gcs
+
+
+class GCSDownloader(DatasetDownloader, protocol="gs://"):
+    """ This class is used to download data from GCS
+    """
+
+    def __init__(self, **kwargs):
+        """
+
+        Args:
+            access_token: Access token to be used to authenticate to
+            unity simulation for downloading the dataset
+        """
+
+    def download(self, source_uri, output, **kwargs):
+        """
+
+        Args:
+            source_uri: This is the downloader-uri that indicates where on
+                GCS the dataset should be downloaded from.
+                The expected source-uri follows these patterns
```

Comment:
It could also be a single file that is zipped

Reply:
Do we have any identifier from cli that indicates whether source GCS location is file or directory? If no then we need to write two seperate flow to download from blob and blobs.

Reply:
I think it should be inferred from the source_uri instead of passing it trough the cli

Reply:
There are so many file compression, also what if there is no indicator for file type(.zip or .*) then how to distinguish. Ex: gs://my-bucket/data, suppose data is file here, how to infer?